packages/starlight/components/Badge.astro

```diff
@@ -0,0 +1,87 @@
+---
+interface Props {
+	theme?: 'blue' | 'pink' | 'green' | 'yellow' | 'purple';
+	text: string;
+}
+const { theme = 'blue', text } = Astro.props;
+---
+
+<span class={theme}>
+	{text}
+</span>
+<style>
+	span {
+		padding-inline: 0.375rem;
+		padding-block: 0.0625rem;
+		border-radius: 0.25rem;
+		font-family: var(--sl-font-system-mono);
+		margin-inline-start: 0.125rem;
+		font-size: var(--sl-text-xs);
+		border: 1px solid;
+		font-weight: normal;
+		color: #fff;
+		background-color: var(--sl-color-bg-badge);
+		border-color: var(--sl-color-border-badge);
+	}
+
+	:global([aria-current='page']) span {
+		background-color: var(--sl-color-text-accent);
+		border-color: var(--sl-color-text-invert);
+		color: var(--sl-color-text-invert);
+	}
+
+	:global([data-theme='dark']) span {
+		--sl-color-bg-badge: hsla(234, 54%, 20%, 1);
+		--sl-color-border-badge: hsla(234, 100%, 60%, 1);
+	}
+
+	span.blue {
+		--sl-color-bg-badge: hsla(234, 80%, 30%, 1);
+		--sl-color-border-badge: hsla(234, 90%, 60%, 1);
+	}
+
+	span.pink {
+		--sl-color-bg-badge: hsla(339, 80%, 30%, 1);
+		--sl-color-border-badge: hsla(339, 90%, 60%, 1);
+	}
+
+	span.green {
+		--sl-color-bg-badge: hsla(102, 80%, 22%, 1);
+		--sl-color-border-badge: hsla(101, 90%, 46%, 1);
+	}
+
+	span.yellow {
+		--sl-color-bg-badge: hsla(40, 80%, 25%, 1);
+		--sl-color-border-badge: hsla(41, 90%, 60%, 1);
+	}
+
+	span.purple {
+		--sl-color-bg-badge: hsla(280, 90%, 30%, 1);
+		--sl-color-border-badge: hsla(281, 90%, 60%, 1);
+	}
+
+	::global([data-theme='dark']) span.blue {
+		--sl-color-bg-badge: hsla(234, 54%, 20%, 1);
+		--sl-color-border-badge: hsla(234, 100%, 60%, 1);
+	}
+
+	::global([data-theme='dark']) span.pink {
+		--sl-color-bg-badge: hsla(339, 39%, 22%, 1);
+		--sl-color-border-badge: hsla(339, 82%, 63%, 1);
+	}
+
+	::global([data-theme='dark']) span.green {
+		--sl-color-bg-badge: hsla(102, 39%, 22%, 1);
+		--sl-color-border-badge: hsla(101, 82%, 63%, 1);
+	}
+
+	::global([data-theme='dark']) span.yellow {
+		--sl-color-bg-badge: hsla(40, 39%, 22%, 1);
+		--sl-color-border-badge: hsla(41, 82%, 63%, 1);
+	}
+
+	::global([data-theme='dark']) span.purple {
+		--sl-color-bg-badge: hsla(280, 39%, 22%, 1);
+		--sl-color-border-badge: hsla(281, 82%, 63%, 1);
+	}
```

Comment:
Shouldn't these be `:global()` and not `::global()`?

```suggestion
	:global([data-theme='dark']) span.blue {
		--sl-color-bg-badge: hsla(234, 54%, 20%, 1);
		--sl-color-border-badge: hsla(234, 100%, 60%, 1);
	}

	:global([data-theme='dark']) span.pink {
		--sl-color-bg-badge: hsla(339, 39%, 22%, 1);
		--sl-color-border-badge: hsla(339, 82%, 63%, 1);
	}

	:global([data-theme='dark']) span.green {
		--sl-color-bg-badge: hsla(102, 39%, 22%, 1);
		--sl-color-border-badge: hsla(101, 82%, 63%, 1);
	}

	:global([data-theme='dark']) span.yellow {
		--sl-color-bg-badge: hsla(40, 39%, 22%, 1);
		--sl-color-border-badge: hsla(41, 82%, 63%, 1);
	}

	:global([data-theme='dark']) span.purple {
		--sl-color-bg-badge: hsla(280, 39%, 22%, 1);
		--sl-color-border-badge: hsla(281, 82%, 63%, 1);
	}
```

Reply:
Woops! I blame VSCode autocomplete 😆